platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -99,41 +99,46 @@ - (instancetype)initWithOptions:(MGLMapSnapshotOptions *)options
         _options = options;
         _loading = false;
         
-        mbgl::DefaultFileSource *mbglFileSource = [MGLOfflineStorage sharedOfflineStorage].mbglFileSource;
-        _mbglThreadPool = mbgl::sharedThreadPool();
-        
-        std::string styleURL = std::string([options.styleURL.absoluteString UTF8String]);
-        
-        // Size; taking into account the minimum texture size for OpenGL ES
-        // For non retina screens the ratio is 1:1 MGLSnapshotterMinimumPixelSize
-        mbgl::Size size = {
-            static_cast<uint32_t>(MAX(options.size.width, MGLSnapshotterMinimumPixelSize)),
-            static_cast<uint32_t>(MAX(options.size.height, MGLSnapshotterMinimumPixelSize))
-        };
-        
-        float pixelRatio = MAX(options.scale, 1);
-        
-        // Camera options
-        mbgl::CameraOptions cameraOptions;
-        if (CLLocationCoordinate2DIsValid(options.camera.centerCoordinate)) {
-            cameraOptions.center = MGLLatLngFromLocationCoordinate2D(options.camera.centerCoordinate);
-        }
-        cameraOptions.angle = MAX(0, options.camera.heading) * mbgl::util::DEG2RAD;
-        cameraOptions.zoom = MAX(0, options.zoomLevel);
-        cameraOptions.pitch = MAX(0, options.camera.pitch);
-        
-        // Region
-        mbgl::optional<mbgl::LatLngBounds> coordinateBounds;
-        if (!MGLCoordinateBoundsIsEmpty(options.coordinateBounds)) {
-            coordinateBounds = MGLLatLngBoundsFromCoordinateBounds(options.coordinateBounds);
-        }
-        
-        // Create the snapshotter
-        _mbglMapSnapshotter = std::make_unique<mbgl::MapSnapshotter>(*mbglFileSource, *_mbglThreadPool, styleURL, size, pixelRatio, cameraOptions, coordinateBounds);
+        [self initMBGLMapSnapshotter:options];
     }
     return self;
 }
 
+- (void)initMBGLMapSnapshotter:(MGLMapSnapshotOptions *)options
```

Comment:
Can we go with a different method name here? Style-wise, I'm not a fan of leading method names with `init` if they're not a proper initializer (and I believe it breaks convention, although I'm in too much of a hurry to dig up a citation)

Reply:
`-configureMapSnapshotterWithOptions:`  perhaps? or `-setUpMapSnapshotterWithOptions:`

Reply:
Good point. Beginning a non-initializer with `init` is problematic because ARC expects to retain its return value. There’s a memorable slide from WWDC11 about a hypothetical [`-copyMachine`](https://download.developer.apple.com/wwdc_2011/adc_on_itunes__wwdc11_sessions__pdf/322_objectivec_advancements_in_depth.pdf#page=18) method that has to be renamed [`-copymachine`](https://download.developer.apple.com/wwdc_2011/adc_on_itunes__wwdc11_sessions__pdf/322_objectivec_advancements_in_depth.pdf#page=24) for the same reason. 😬